Salesforce 10 User Guide (#576)
mulesoft/docs-connectors: modules/ROOT/pages/salesforce/salesforce-connector-100-design-center.adoc

= Salesforce Design Center Configuration - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

Design Center enables you to create apps visually. To use Design Center, work with your Anypoint Platform administrator to ensure that you have a xref:access-management::environments.adoc#to-create-a-new-environment[Design environment]. For more information, see the xref:design-center::fd-tour.adoc[Flow Designer Tour].

> To create an app in Design Center:

Reviewer: ```suggestion
To create an app in Flow Designer:
```

* Configure the input source (trigger) for your app.
* Add the connector as a component to process the input for the app.

> For information about Design Center field values, see the xref:salesforce/salesforce-connector.adoc[Salesforce Connector Reference] list for this Salesforce version.

Reviewer: ```suggestion
For information about Flow Designer field values, see the xref:salesforce/salesforce-connector.adoc[Salesforce Connector Reference] list for this Salesforce version.
```

| <<Configure OAuth SAML Authorization, OAuth SAML>> |https://help.salesforce.com/apex/HTViewHelpDoc?id=remoteaccess_oauth_SAML_bearer_flow.htm[OAuth 2.0 SAML Bearer Assertion Flow]
| <<Configure OAuth Username Password Authorization, OAuth Username Password>> |https://help.salesforce.com/articleView?id=remoteaccess_oauth_username_password_flow.htm[OAuth 2.0 Username-Password Flow]
|===
+
. Specify the parameters as required for each configuration type, and click *Test* to ensure you have a valid connection to Salesforce.
. If needed, configure the *Apex* tab to decide what metadata to fetch and to add Apex class names.
. If you haven't specified a `Connection Type`, and if needed, configure the *Advanced* tab to set the `Max Idle Time` and `Time Units` for how long to hold open a connection before it expires.
. After you specify a connection type, if needed, you can use the *Advanced* tab to change values for the Reconnection Strategy, Pooling Profile, and Expiration Policy.
. If needed, set the `Redelivery Policy` on the main configuration screen. This indicates the maximum number of tries to deliver trigger content.

== Configure the Target Component

> . Click “+” next to the trigger card.

Reviewer: ```suggestion
. In Flow Designer, click “+” next to the trigger card.
```

. In *Select a component*, search for the connector name.
. Select the connector name.
. Select an operation for the connector.

> . Enter the required values in the General tab.

Reviewer: ```suggestion
. Enter the required values in the **General** tab.
```

. If needed, enter values for other tabs.
. Specify access information to the connector resource.
. If a *Test* button is available, click it to test the connection.

> In Design Center's Flow Designer, after you specify a trigger, you can add the Salesforce connector as a component to process the input you received from the trigger. You may also want to add a DataWeave Transform between the trigger and the component to transform the input information to what you want as input to the Salesforce connector component. The Salesforce connector doesn't have to be a component, it depends on what you're trying to accomplish with your app.

Reviewer: I'm not sure about using "the Salesforce connector"  Didn't Kim say we shouldn't use that form? Rather just Salesforce Connector.   ?

> The following are examples of the operations you can set in the Salesforce component:

Reviewer: Now calling it the Salesforce component... I don't think we should mix terminology.

> The following are examples of the operations you can set in the Salesforce component:

Reviewer: ```suggestion
The following are examples of the operations you can set in Salesforce Connector:
```

[%header,cols="30s,70a"]
|===
|Operation |Use
|Create |

* *Type*: Salesforce object type.
* *Records*: Create a collection of Salesforce objects.
|Query |

* *Salesforce query*: Salesforce query to retrieve objects.
* *Parameters*: Values for placeholders in the Salesforce query.
|Update |

* *Type*: Salesforce object type.
* *Records*: Produce a collection of Salesforce objects to be updated.
|Delete |

* *Records To Delete Ids*: Produce a collection of Salesforce objects to be deleted.
|===

== Configure Basic Authentication Authorization

Important fields:

* *Username* (required): Enter the Salesforce username.
* *Password* (required): Enter the corresponding password.
* *Security Token*: Enter the corresponding security token.
* *Authorization URL*: Enter the URL for the endpoint configured to handle SOAP authentication requests.

[NOTE]

> Implementing OAuth 2.0-based authentication mechanisms involves extra steps, but may be preferred if your service is exposed to external users, as it ensures better security.

Reviewer: ```suggestion
Implementing OAuth 2.0-based authentication mechanisms involves extra steps, but it ensures better security if your service is exposed to external users.
```

Author: Used "offers" instead of "ensures" re Kim.
